acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+    """There should be at least 1 subnet available once bootstrap is completed.
+       By default 4 subnets should be shown under an AWS controller."""
+    subnets_dict = get_subnets(client)[0]
+    if not subnets_dict:
+        raise JujuAssertionError('No subnet can be Found.')
+    else:
+        log.info('{} subnet(s) have been found.'.format(str(len(subnets_dict))))
+
+
+def add_space_with_existing_subnet(client, space_name):
+    # Run juju add-space <space_name> <subnet_cidr>
+    subnets_cidr_list = get_subnets(client)[1]
+    subnet_cidr = subnets_cidr_list[0]
+    # Bug 1704105, merge_stderr=True is required.
+    client.get_juju_output(
+        'add-space', space_name, subnet_cidr, include_e=False, merge_stderr=True)
+    return (space_name, subnet_cidr)
+
+
+def assert_added_space(client):
+    """Validate if configurations in new added space are as expected.
+       Five checkpoints:
+       1. After new space is added, juju spaces --format json should give output
+          in json format, in comparison to no space exists.
+       2. Total number of space should be 1.
+       3. The name of new added space should be the same as specified.
+       4. Total number of subnet CIDR from this new added space should be 1.
+       5. The CIDR data from new added space should be the same as specified.
+    """
+    space_name, subnet_cidr = add_space_with_existing_subnet(
+        client, space_name='testspace')
+    # New added space should exist at this point,
+    # set merge_stderr to False to eliminate noise in output.
+    raw_output = client.get_juju_output(
+        'spaces', '--format', 'json', include_e=False, merge_stderr=False)
+    try:
+        space_output = json.loads(raw_output)
+    except ValueError as e:
+        log.error('New added space cannot be found.')
+        raise e
+    # These 4 validations are in sequence.
+    if len(space_output['spaces'].keys()) != 1:
+        raise JujuAssertionError(
+            'Incorrect number of space(s). Found: {}; Expected: 1.'.format(
+            str(len(space_output['spaces'].keys()))))
+    elif space_output['spaces'].keys()[0] != space_name:
+        raise JujuAssertionError(
+            'Incorrect space name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'].keys()[0], space_name))
+    elif len(space_output['spaces'][space_name].keys()) != 1:
+        raise JujuAssertionError(
+            'Incorrect number of CIDR(s). Found: {}; Expected: 1.'.format(
+            str(len(space_output['spaces'][space_name].keys()))))
+    elif space_output['spaces'][space_name].keys()[0] != subnet_cidr:
+        raise JujuAssertionError(
+            'Incorrect CIDR name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'][space_name].keys()[0], subnet_cidr))
+    else:
+        log.info('New added space {} has been validated.'.format(space_name))
+
+
+def assert_app_status(client, charm_name, expected):
+    # Run juju status --format json
```

Comment:
What is this asserting about the app status exactly? Docstring would cover this etc.